cmd/snap-update-ns/user.go

```diff
@@ -78,18 +84,44 @@ func (ctx *UserProfileUpdateContext) LoadDesiredProfile() (*osutil.MountProfile,
 	return profile, nil
 }
 
+// SaveCurrentProfile does nothing at all.
+//
+// Per-user mount profiles are not persisted yet.
+func (ctx *UserProfileUpdateContext) SaveCurrentProfile(profile *osutil.MountProfile) error {
+	return nil
+}
+
+// LoadCurrentProfile returns the empty profile.
+//
+// Per-user mount profiles are not persisted yet.
+func (ctx *UserProfileUpdateContext) LoadCurrentProfile() (*osutil.MountProfile, error) {
+	return &osutil.MountProfile{}, nil
+}
+
 func applyUserFstab(ctx MountProfileUpdateContext) error {
 	desired, err := ctx.LoadDesiredProfile()
 	if err != nil {
 		return err
 	}
 	debugShowProfile(desired, "desired mount profile")
+
+	current, err := ctx.LoadCurrentProfile()
+	if err != nil {
+		return err
+	}
+	debugShowProfile(current, "current mount profile")
+
 	as := ctx.Assumptions()
-	_, err = applyProfile(ctx, &osutil.MountProfile{}, desired, as)
+	_, err = applyProfile(ctx, current, desired, as)
 	return err
 }
 
 // desiredUserProfilePath returns the path of the fstab-like file with the desired, user-specific mount profile for a snap.
 func desiredUserProfilePath(snapName string) string {
 	return fmt.Sprintf("%s/snap.%s.user-fstab", dirs.SnapMountPolicyDir, snapName)
 }
+
+// currentUserProfilePath returns the path of the fstab-like file with the applied, user-specific mount profile for a snap.
+func currentUserProfilePath(snapName string, uid int) string {
```

Comment:
How/where will this be used? And why is it different from desiredUserProfilePath which does not take a uid ? Mostly wondering if this should be part of a large function or not - its so trivial it seems we might not need it as a separate helper(?). Or maybe it should be a method on ctx *UserProfileUpdateContext ? But only more context can tell :)

Reply:
This is used in the subsequent patches where we actually start saving the user profile. It is different from desired because the desired profile is a template where various variables derived from the UID are used (.e.g. SNAP_USER_DATA or XDG_RUNTIME_DIR). The current profile has an UID because those variables are already derived for that user there.

The desired mount profile may look like this:

```
tmpfs $SNAP_USER_DATA/foo ...
```

A current mount profile for user 1000 might look like:

```
tmpfs /home/zyga/snap/my-fancy-snap/42/foo ...
```

We need to have those separately because per-user mount namespaces are updated one-by-one, for each one that exists.